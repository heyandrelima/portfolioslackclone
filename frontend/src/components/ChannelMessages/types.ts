import { Message } from '../../types/api';

export type Props = {
    messages: Message[];
}
