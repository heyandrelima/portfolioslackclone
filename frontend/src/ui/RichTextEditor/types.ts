export type ContainerProps = {
    menuHeight: number;
}