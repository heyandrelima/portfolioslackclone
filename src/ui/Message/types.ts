import { Message } from '../../types/api';

export type Props = {
    message: Message
}
