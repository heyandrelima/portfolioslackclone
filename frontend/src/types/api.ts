export type Author = {
    name: string;
    avatar: string;
}

export type Message = {
    createdAt: Date;
    id: string;
    author: Author;
    body: string;
}

export type Channel = {
    id: string;
    label: string;
    description?: string;
    messages?: Array<Message>;
}
