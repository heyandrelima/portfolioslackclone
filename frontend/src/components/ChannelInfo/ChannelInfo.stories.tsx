import ThemeContext from '../../context/Theme';
import { defaultTheme } from '../../themes';
import { Channel } from '../../types/api';
import ChannelInfo from './ChannelInfo';

const Story = {
    title: 'Components/ChannelInfo',
    component: ChannelInfo,
};

export const Default = () => {
    const channel: Channel = {
        id: 'abc123',
        label: 'announcements',
        description: 'New stuff coming',
    };
    return (
        <ThemeContext theme={defaultTheme}>
            <ChannelInfo channelInfo={channel} />
        </ThemeContext>
    );
};

export default Story;
