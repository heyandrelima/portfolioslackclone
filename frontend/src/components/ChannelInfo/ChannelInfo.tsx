import styled from "styled-components";
import { DefaultStyledComponentProps } from "../../themes";
import { Channel } from "../../types/api";

type Props = {
  channelInfo: Channel;
};

const ChannelInfo = ({ channelInfo: { label, description } }: Props) => (
  <Container data-test-id="channel-info">
    <Label>#{label}</Label>
    {description && <Description>{description}</Description>}
  </Container>
);

const Container = styled.div`
  background: #fff;
  width: 100$;
  border-bottom: 1px solid #ccc;
  padding: 0.6em;
`;

const Label = styled.h2<DefaultStyledComponentProps>`
  font-family: ${(props: DefaultStyledComponentProps) =>
    props.theme.fontFamily};
  color: ${(props: DefaultStyledComponentProps) => props.theme.fontColorDarker};
  font-size: 1.2em;
  font-weight: bold;
  margin: 0.4em 0;
  cursor: pointer;
`;

const Description = styled.p`
  font-family: ${(props: DefaultStyledComponentProps) =>
    props.theme.fontFamily};
  color: ${(props: DefaultStyledComponentProps) => props.theme.fontColorDark};
  font-size: 0.9em;
  margin: 0 0 0.4em;
`;

export default ChannelInfo;
