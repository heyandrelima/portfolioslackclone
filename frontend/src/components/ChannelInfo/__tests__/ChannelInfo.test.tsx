import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import ChannelInfo from '../ChannelInfo';
import { Channel } from '../../../types/api';

describe('components/ChannelInfo', () => {
    it('shoud render correctly', () => {
        const channel: Channel = {
            id: 'abc123',
            label: 'announcements',
            description: 'New stuff coming',
        };
        const component = render(<ChannelInfo channelInfo={channel} />).queryByTestId('channel-info');
        expect(component).toMatchSnapshot();
    });
});
