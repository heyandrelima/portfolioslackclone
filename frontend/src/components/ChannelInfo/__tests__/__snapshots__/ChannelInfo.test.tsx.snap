// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ChannelInfo shoud render correctly 1`] = `null`;

exports[`components/ChannelMessages shoud render correctly 1`] = `null`;
